core/src/editor.rs

```diff
@@ -124,7 +125,14 @@ impl<'a> MapManager<'a> {
         transform.translation.y = position.y;
     }
     fn delete_element(&mut self, entity: Entity) {
-        self.entities.kill(entity);
+        if let Some(element_kill_callback) = self.element_kill_callbacks.get(entity) {
+            let system = element_kill_callback.system.clone();
+            self.commands
+                .add(move |world: &World| (system.lock().unwrap().run)(world).unwrap());
```

Comment:
This was causing a panic because it wasn't actually deleting the element anymore, just the fish.

We could either make the kill callback responsible for actually killing the element entity, or better than that I think, we could just run the kill callback first, and then delete the element entity like this:

```suggestion
            let system = element_kill_callback.system.clone();
            self.commands.add(move |world: &World| {
                // Run the kill callback first
                (system.lock().unwrap().run)(world).unwrap();

                // Then delete the entity
                let entities = world.get_resource::<Entities>().unwrap();
                let mut entities = entities.borrow_mut();
                entities.kill(entity);
            });
```

Reply:
This causes an additional search of the entities that would not be necessary if we kill the element as part of its kill callback function. I think it's because your other change removes the kill of the entity.

Reply:
I just confirmed the panic is due to the fact that the FishSchool still exists with a vector of Fish entities, but since they have been killed they are no longer returning back the expected Transforms (for moving the fish around). I would think that it would make more sense if the kill callback did all of the work of killing the element (if it exists for the entity). If you really want to kill the parent at the end, maybe we should rename the component to something akin to "dispose callback", "before kill callback", or "prekill callback". Thoughts?

Reply:
> I think it's because your other change removes the kill of the entity.

Oh, yeah, duh. I just missed that, LOL. Good catch. :)

> I would think that it would make more sense if the kill callback did all of the work of killing the element (if it exists for the entity).

If that makes sense to you, I'm fine with that. I could go either way, I just didn't want it to be confusing.